General

JACKAR710 opened the pull request:
- Pinmap.h aangemaakt
     - Pins en datadirection defined
- millis() functie gemaakt
- Appinfo.h toegevoegd

JACKAR710 committed:
Added:
- Millis function
- Appinfo

JACKAR710 committed:
Pinmap.h:
Pins and data direction defined.

Someone-s-out-there commented on RobotCar/Functions/Millis/Millis.c:
```diff
@@ -0,0 +1,16 @@
+
+unsigned long millis()
+{
+	clock_t runTimeInMS = clock() / (CLOCKS_PER_SEC / 1000);
```
waar komt dit vandaan

Someone-s-out-there replied:
waar komt dit vandaan

Someone-s-out-there commented on RobotCar/Functions/Pinmap.h:
```diff
@@ -0,0 +1,63 @@
+		(OUTPUT << LED_LEFT) |
+		(OUTPUT << LED_RIGHT);
+		
+DDRC |= (INPUT << SW1) |
```
Gaat waarschijnlijk niet werken

Someone-s-out-there replied:
al deze assingments moeten in een functie

Someone-s-out-there reviewed (changes requested):
Inputs gaan waarschijnlijk niet werken.

JACKAR710 committed:
Millis function working.

JACKAR710 committed:
Millis function fixed and working

Someone-s-out-there commented:
De ISR hoeft niet in de header te staan

Someone-s-out-there commented on RobotCar/Functions/Millis/Millis.h:
```diff
@@ -0,0 +1,23 @@
+
+void millis_init(void);
+
+ISR(TIMER0_COMPA_vect);
```
hoeft niet in de header

Someone-s-out-there commented on RobotCar/Functions/Millis/Millis.c:
```diff
@@ -0,0 +1,50 @@
+
+// Interrupt Service Routine that is automatically executed as soon as
+// timer/counter 0 has reached its compare value
+ISR(TIMER0_COMPA_vect)
```
probeer de ISR's onder aan de bestanden te zetten

Someone-s-out-there reviewed (changes requested):
voor changes zie de comments

JACKAR710 committed:
Buttons alles gaat fout

21 errors

Someone-s-out-there commented on RobotCar/Functions/Buttons/mode_select.c:
```diff
@@ -0,0 +1,69 @@
+
+#define READSW1() PINC&~(1 << SW1)
+
+millis_init();
```
?? code kan niet geexcuteerd worden zonder dat het in eenfuctie staat

Someone-s-out-there commented on RobotCar/Functions/Buttons/mode_select.c:
```diff
@@ -0,0 +1,69 @@
+
+millis_init();
+
+sei();
```
?? code kan niet geexcuteerd worden zonder dat het in eenfuctie staat

Someone-s-out-there reviewed (changes requested):
Heb jij geen rode squiggly lines in je editor?

Someone-s-out-there commented on RobotCar/Functions/Buttons/mode_select.c:
```diff
@@ -0,0 +1,69 @@
+/*
+ * CFile1.c
```
dit lijkt incorrect

JACKAR710 committed:
Succesful build
0 Errors
0 Warnings
Het idee is dat er na 2 keer op de knop te drukken pin PB4 een output geeft, maar het werkt niet.

JACKAR710 committed:
Working button cycle

JACKAR710 committed:
working button

Someone-s-out-there commented on RobotCar/Functions/Millis/Millis.c:
```diff
@@ -8,24 +8,29 @@
+		OCR0A = 249;
+		
+		// Timer/Counter0 Output Compare Match A Interrupt Enable
+		TIMSK0 |= (1<<OCIE0A);
```
millisEnabled nog op 1 zetten

Someone-s-out-there commented on RobotCar/Functions/Pinmap.c:
```diff
@@ -0,0 +1,36 @@
+
+#include "Pinmap.h"
+
+void registers_init(void)
```
kan beter IO_init(void) zijn

Someone-s-out-there replied:
gezien hier alleen de IO register worden geinit
```suggestion
void IO_init(void)
```

Someone-s-out-there reviewed (changes requested):
zie comments voor changes

Someone-s-out-there commented on RobotCar/Functions/Buttons/mode_select.c:
```diff
@@ -0,0 +1,63 @@
+ */ 
+#include "mode_select.h"
+
+#define READSW1 !(PINC & (1 << SW1))
```
needs Docs + function like macro
`READSW1() `

Someone-s-out-there replied:
```suggestion
// pressed =1 else 0
#define READSW1() !(PINC & (1 << SW1)) 
```

Someone-s-out-there commented on RobotCar/Functions/Buttons/mode_select.c:
```diff
@@ -0,0 +1,63 @@
+			}
+
+			switch (switchSelect) {
+				case 0:
```
enum for swich case

Someone-s-out-there replied:
of #defined constants

JACKAR710 committed:
Working mode select

JACKAR710 committed:
Sjoerd blij?

Someone-s-out-there committed:
Merge branch 'dev' into General
